drivers/overlay/overlay.go

```diff
@@ -882,6 +882,17 @@ func (d *Driver) UpdateLayerIDMap(id string, toContainer, toHost *idtools.IDMapp
 	if err := idtools.MkdirAs(diffDir, 0755, rootUID, rootGID); err != nil {
 		return err
 	}
+
+	if rootUID != 0 {
+		parent := filepath.Dir(diffDir)
+		for parent != "/" {
+			err = os.Chmod(parent, 0755)
```

Comment:
Wouldn't 711 be enough.  Don't directories have to be searchable not listable?

Reply:
yes that is enough, I have changed the patch to use 0711.

Also, I have added another check to not override the mode if it is already permissive